src/node/internal/internal_dns.ts

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) 2017-2022 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+// Copyright Joyent and Node contributors. All rights reserved. MIT license.
+
+import { sendDnsRequest } from 'node-internal:internal_dns_client';
+import { DnsError } from 'node-internal:internal_errors';
+import { validateString } from 'node-internal:validators';
+import * as errorCodes from 'node-internal:internal_dns_constants';
+import { default as dnsUtil } from 'node-internal:dns';
+
+export function getServers(): string[] {
+  throw new Error('Not implemented');
+}
+
+export function lookup(): void {
+  throw new Error('Not implemented');
+}
+
+export function lookupService(): void {
+  throw new Error('Not implemented');
+}
+
+export function resolve(): void {
+  throw new Error('Not implemented');
+}
+
+export function resolve4(): void {
+  throw new Error('Not implemented');
+}
+
+export function resolve6(): void {
+  throw new Error('Not implemented');
+}
+
+export function resolveAny(): void {
+  throw new Error('Not implemented');
+}
+
+export function resolveCname(): void {
+  throw new Error('Not implemented');
+}
+
+export function resolveCaa(
+  name: string
+): Promise<
+  { critical: number; issue?: string; iodef?: string; issuewild?: string }[]
+> {
+  validateString(name, 'name');
+  return sendDnsRequest(name, 'CAA').then((json) => {
```

Comment:
Why not async await syntax?

Reply:
Because `validateString()` should throw synchronously whereas the DNS request errors should be async errors.

Reply:
A comment along those lines would be helpful